spyder/plugins/updatemanager/workers.py

```diff
@@ -249,17 +278,9 @@ def _download_installer(self):
 
     def _clean_installer_path(self):
         """Remove downloaded file"""
-        if osp.exists(self.installer_path):
-            try:
-                shutil.rmtree(self.installer_path)
-            except OSError as err:
-                logger.debug(err, stack_info=True)
-
-        if osp.exists(self.installer_size_path):
-            try:
-                shutil.rmtree(self.installer_size_path)
-            except OSError as err:
-                logger.debug(err, stack_info=True)
+        installer_dir = osp.dirname(self.installer_path)
+        if osp.exists(installer_dir):
+            shutil.rmtree(installer_dir)
```

Comment:
Two questions about this:

* Do we want to leave this without a try/except?
* Does `installer_dir` is created again when a new installer is downloaded?

Reply:
Sorry, I completely spaced it there. I meant to put that in a try statement. Will fix.

So the `install_dir` should be: `get_temp_dir()/updates/<version>`. Only the `<version>` directory (and contents) will be deleted and recreated if the installer is downloaded again. However, it should only be removed if there is an error in downloading. If the download is successful, the directory and contents should persist until the system (or user) purges the temporary directories (e.g. a reboot). Different versions of the installers can be downloaded and coexist; any errors and accompanying purges will only affect the version currently being considered.